cpp/src/io/comp/statistics.cu

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "gpuinflate.hpp"
+
+#include <rmm/exec_policy.hpp>
+#include <thrust/transform_reduce.h>
+
+namespace cudf::io {
+
+writer_compression_statistics collect_compression_statistics(
```

Comment:
Why not `shared_ptr` anymore?

Reply:
This function creates a `writer_compression_statistics object`, no need to know how its lifetime might be managed after it's created. If we think of this as a factory, we could return a `unique_ptr`, but I don't think it's beneficial here.